sdk/src/main/java/com/amplitude/experiment/storage/Cache.kt

```diff
@@ -0,0 +1,66 @@
+package com.amplitude.experiment.storage
+
+import com.amplitude.experiment.util.toMap
+import org.json.JSONObject
+
+internal class LoadStoreCache<V>(
+    private val namespace: String,
+    private val storage: Storage,
+    private val transformer: ((value: Any) -> V)? = null
+) {
+    private val cache: MutableMap<String, V> = mutableMapOf()
+
+    fun get(key: String): V? = synchronized(cache) {
+        return cache[key]
+    }
+
+    fun getAll(): Map<String, V> = synchronized(cache) {
+        return HashMap(cache)
+    }
+
+    fun put(key: String, value: V) = synchronized(cache) {
+        cache[key] = value
+    }
+
+    fun putAll(values: Map<String, V>) = synchronized(cache) {
+        cache.putAll(values)
+    }
+
+    fun remove(key: String) = synchronized(cache) {
+        cache.remove(key)
+    }
+
+    fun clear() = synchronized(cache) {
+        cache.clear()
+    }
+
+    fun load() = synchronized(cache) {
+        val rawValues = storage.get(namespace)
+        val jsonValues: Map<String, Any?> = try {
+            JSONObject(rawValues).toMap()
```

Comment:
Looking at this from my IDE, rawValues may be null, which will cause the JSONObject constructor to throw. I see this is wrapped in a try catch, but best to be explicit, and use the api as intended. 

Try to handle the null rawValues explicitly rather than relying on a throw.